lib/model/model.go

```diff
@@ -492,8 +485,17 @@ func (m *model) restartFolder(from, to config.FolderConfiguration, cacheIgnoredF
 	m.fmut.Lock()
 	defer m.fmut.Unlock()
 
+	// Cache the (maybe) existing fset before it's removed by cleanupFolderLocked
+	fset := m.folderFiles[to.ID]
+
 	m.cleanupFolderLocked(from)
 	if !to.Paused {
+		if fset == nil {
+			// Create a new fset. Might take a while and we do it under
+			// locking, but it's unsafe to create fset:s concurrently so
+			// that's the price we pay.
+			fset = db.NewFileSet(to.ID, to.Filesystem(), m.db)
```

Comment:
Actually we could maybe release `fmut` for the duration since we have the restart lock, and are guaranteed that only CommitConfiguration could run at any given time ... maybe

Reply:
Are we guaranteed that only a single CommitConfiguration can be running at a time? I don't think it is, and that's the only way how fset could ever be nil. If e.g. a folder config change races with the same folder being paused afterwards, and somehow the pausing gets ahead of the change, then we might try to restart a folder that's already stopped. (we should base the "from-state" on the internal state, not the `from` config given to CommitConfiguration to avoid such problems, but that's a different matter)

Reply:
No, actually we're not guaranteed that since it's called from cfg.Replace() which is for example called from inside ClusterConfig(). Potentially we could expand the restartMut to cover all config changes for a folder (currently it doesn't) for the meantime I'm happier keeping this inside fmut